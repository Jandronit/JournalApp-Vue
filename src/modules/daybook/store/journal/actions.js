import journalApi from '@/api/journalApi';

// export const myAction = async ({ commit }) => {
//   // do something
// }

export const loadEntries = async ( { commit } ) => {
    const { data } = await journalApi.get('/entries.json')
    if ( !data ) {
        commit('setEntries', []);
        return
    }
    const entries = []
    for ( let id of Object.keys( data ) ) {
        entries.push({
            id,
            ...data[id]
        })
    }
    commit('setEntries', entries)
}

export const updateEntry = async ({ commit }, entry) => {
    const { date, picture, text } = entry
    const dataToSave = { date, picture, text }

    const resp = await journalApi.put(`/entries/${ entry.id }.json`, dataToSave)

    resp.status === 200 && commit('updateEntry', { ...entry })
}

export const createEntry = async ({ commit }, entry) => {
    const { date, picture, text } = entry
    const dataToSave = { date, picture, text }

    const resp = await journalApi.post(`/entries.json`, dataToSave)

    resp.status === 200 && commit('addEntry', { ...dataToSave, id: resp.data.name })

    return resp.data.name
}

export const deleteEntry = async ({ commit }, id) => {
    const resp = await journalApi.delete(`/entries/${ id }.json`)
    resp.status === 200 && commit('deleteEntry', id)

    return id
}
