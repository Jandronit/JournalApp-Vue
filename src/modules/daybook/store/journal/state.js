
export default () =>  ({
    isLoading: true,
    entries: [
        {
            id: new Date().getTime(),
            date: new Date().toDateString(),
            text: 'lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, quod.',
            picture: null,
        },
        {
            id: new Date().getTime() + 1000,
            date: new Date().toDateString(),
            text: 'Voluptate culpa quia, quod, quas, voluptatum quibusdam quidem voluptates quae quos voluptas quas.',
            picture: null,
        },
        {
            id: new Date().getTime() + 2000,
            date: new Date().toDateString(),
            text: 'Fugiat consequuntur, quod, quas, voluptatum quibusdam quidem voluptates quae quos voluptas quas.',
            picture: null,
        }
    ]
})
