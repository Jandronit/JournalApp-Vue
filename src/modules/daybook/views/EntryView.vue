<template>
<div class="entry-title d-flex justify-content-between p-2">

  <div>
    <span class="text-success fs-3 fw-bold">11</span>
    <span class="mx-1 fs-3">Diciembre</span>
    <span class="mx-2 fs-4 fw-light">2022</span>
  </div>

  <div>
    <button class="btn btn-danger mx-2">
      Borrar
      <i class="fas fa-trash-alt"></i>
    </button>
    <button class="btn btn-primary">
      Subir foto
      <i class="fas fa-upload"></i>
    </button>
  </div>

</div>

  <hr>
  <div class="d-flex flex-column px-3 h-75">
    <textarea placeholder="¿Qué sucedió ahora"></textarea>
  </div>
  <Fab></Fab>

  <img src="https://images.pexels.com/photos/2662116/pexels-photo-2662116.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1" alt="entry-picture" class="img-thumbnail">
</template>

<script>
import {defineAsyncComponent} from 'vue';

export default {
  name: "EntryView",
  components: {
    Fab: defineAsyncComponent(() => import('../components/Fab.vue')),
  }
}
</script>

<style lang="scss" scoped>

textarea{
  font-size: 20px;
  border: none;
  height: 100%;

  &:focus{
    outline: none;
  }
}

img{
  width: 12.5rem;
  position: fixed;
  bottom: 9.375rem;
  right: 1.25rem;
  box-shadow: 0px 5px 10px rgba($color: #000000, $alpha: 0.2);}
</style>
