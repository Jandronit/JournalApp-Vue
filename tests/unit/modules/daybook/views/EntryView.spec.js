import { shallowMount } from "@vue/test-utils";
import { createStore } from "vuex";

import journal from "@/modules/daybook/store/journal";
import EntryView from "@/modules/daybook/views/EntryView.vue";

import { journalState } from "../../../../mocks/test-journal-state";

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });
describe("Test in component EntryView", () => {
  const store = createVuexStore(journalState);
  const mockRouter = {
    push: jest.fn(),
  };

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryView, {
      props: {
        id: journalState.entries[0].id,
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });

  test("You must remove the user because the id does not exist", () => {
    shallowMount(EntryView, {
      props: {
        id: "ID not exists in the store",
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
    expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" });
  });

  test("You must show the entry", () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(mockRouter.push).not.toHaveBeenCalled();
  });
});
