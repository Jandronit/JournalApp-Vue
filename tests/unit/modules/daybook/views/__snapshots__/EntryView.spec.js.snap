// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component EntryView You must show the entry 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">13</span><span class="mx-1 fs-3">Marzo</span><span class="mx-2 fs-4 fw-light">2023, Lunes</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fas fa-trash-alt"></i></button><button class="btn btn-primary"> Subir foto <i class="fas fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Qué sucedió ahora"></textarea></div>
<!--v-if-->
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
